launchpad/src/launchpad.rs

```diff
@@ -3,324 +3,340 @@
-                    nr_winning_tickets,
-                });
-            }
-            OperationCompletionStatus::Completed => {
-                self.start_confirmation_period(VEC_MAPPER_START_INDEX, nr_winning_tickets);
-            }
-        };
-
-        Ok(run_result.output_bytes().into())
-    }
-
     #[payable("*")]
     #[endpoint(confirmTickets)]
     fn confirm_tickets(
         &self,
         #[payment_token] payment_token: TokenIdentifier,
         #[payment_amount] payment_amount: Self::BigUint,
-        nr_tickets_to_confirm: usize,
     ) -> SCResult<()> {
-        self.require_stage(LaunchStage::ConfirmTickets)?;
+        self.require_confirmation_period()?;
+        self.require_launchpad_tokens_deposited()?;
 
         let caller = self.blockchain().get_caller();
         require!(
-            !self.blacklist().contains(&caller),
+            !self.is_user_blacklisted(&caller),
             "You have been put into the blacklist and may not confirm tickets"
         );
+        require!(
+            !self.has_user_confirmed_all_tickets(&caller),
+            "You already confirmed all tickets"
+        );
 
         let ticket_payment_token = self.ticket_payment_token().get();
         let ticket_price = self.ticket_price().get();
-        let total_ticket_price = Self::BigUint::from(nr_tickets_to_confirm) * ticket_price;
+
+        let nr_tickets = self.get_total_number_of_tickets_for_address(&caller);
+        let total_ticket_price = Self::BigUint::from(nr_tickets) * ticket_price;
 
         require!(
             payment_token == ticket_payment_token,
             "Wrong payment token used"
         );
-        require!(payment_amount == total_ticket_price, "Wrong amount sent");
-
-        let (first_ticket_id, last_ticket_id) = self.try_get_ticket_range(&caller)?;
-        let nr_tickets = last_ticket_id - first_ticket_id + 1;
-
-        require!(
-            nr_tickets >= nr_tickets_to_confirm,
-            "Trying to confirm too many tickets"
-        );
-
-        let current_generation = self.current_generation().get();
-        let winning_tickets = self.get_tickets_with_status(
-            &caller,
-            TicketStatus::Winning {
-                generation: current_generation,
-            },
-        );
-
         require!(
-            nr_tickets_to_confirm <= winning_tickets.len(),
-            "Trying to confirm too many tickets"
+            payment_amount == total_ticket_price,
+            "Wrong amount sent, must confirm all tickets"
         );
 
-        for winning_ticket_id in &winning_tickets[..nr_tickets_to_confirm] {
-            self.ticket_status(*winning_ticket_id)
-                .set(&TicketStatus::Confirmed {
-                    generation: current_generation,
-                });
-        }
-
-        self.total_confirmed_tickets()
-            .update(|confirmed| *confirmed += nr_tickets_to_confirm);
+        self.confirmed_all_tickets(&caller).set(&true);
```

Comment:
as a user I can confirm less tokens than all. add back the old code here please.